### PR TITLE
chore: KT-45550 lazy init in JS IR

### PR DESCRIPTION
**Pull request by p7nov**
@dromanov-jb please check

### CODE REVIEW

File: docs/topics/js/js-ir-compiler.md [hunk 1]
```diff
@@ -38,9 +38,8 @@ This behaviour is overwritten by any settings in the `build.gradle(.kts)`, howev
 
 ## Ignoring compilation errors
 
->_Ignore compilation errors_ mode is [Experimental](components-stability.md). It may be dropped or changed
-> at any time. Opt-in is required (see details below). Use it only for evaluation purposes. We appreciate your feedback 
-> on it in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
+>_Ignore compilation errors_ mode is [Experimental](components-stability.md). It may be dropped or changed at any time.
+> Opt-in is required (see the details below), and you should use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT.
```
Comment (koshachy) [inline docs/topics/js/js-ir-compiler.md:42]:
```suggestion
> Opt-in is required (see the details below), and you should use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
```

File: docs/topics/js/js-ir-compiler.md [hunk 2]
```diff
@@ -72,6 +71,37 @@ kotlin {
...
+## Lazy initialization of top-level properties
+
+> Lazy initialization of top-level properties is [Experimental](components-stability.md). It may be dropped or changed at any time.
+> Opt-in is required (see the details below), and you should use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-44320).
+>
+{type="warning"}
+
+For better application startup performance, the Kotlin/JS IR compiler offers an option to initialize top-level properties
+lazily. This way, the application loads without initializing all the top-level properties used in its code. It initializes
+only the ones needed at startup; other properties get initialized later when the code that uses them is executed.
```
Comment (koshachy) [inline docs/topics/js/js-ir-compiler.md:83]: The second part of the sentence is too heavy. Could we get rid of this passive voice?

Sorry, can't find a better solution :/

File: docs/topics/js/js-ir-compiler.md [hunk 3]
```diff
@@ -72,6 +71,37 @@ kotlin {
...
+> Lazy initialization of top-level properties is [Experimental](components-stability.md). It may be dropped or changed at any time.
+> Opt-in is required (see the details below), and you should use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-44320).
+>
+{type="warning"}
+
+For better application startup performance, the Kotlin/JS IR compiler offers an option to initialize top-level properties
+lazily. This way, the application loads without initializing all the top-level properties used in its code. It initializes
+only the ones needed at startup; other properties get initialized later when the code that uses them is executed. 
+
+We're still working on this feature, so it's experimental and requires an opt-in. To use the lazy initialization of top-level
```
Comment (koshachy) [inline docs/topics/js/js-ir-compiler.md:85]: IDK do we really need to highlight that we are still working on it?
Maybe we should tell it in the components-stability.md doc page (experimental section)?

File: docs/topics/js/js-ir-compiler.md [hunk 4]
```diff
@@ -72,6 +71,37 @@ kotlin {
...
+properties, add the `-Xir-property-lazy-initialization` compiler option when compiling the code with the JS IR compiler.
```
Comment (koshachy) [inline docs/topics/js/js-ir-compiler.md:86]: Maybe too much "compilers" for a single sentence :)
But it seems justified.

File: docs/topics/js/js-ir-compiler.md [hunk 5]
```diff
@@ -72,6 +71,37 @@ kotlin {
...
+{type="warning"}
+
+The Kotlin/JS IR compiler is capable of generating TypeScript definitions from your Kotlin code. These definitions can be
+used by JavaScript tools and IDEs when working on hybrid apps to provide autocompletion, support static analyzers, and
+make it easier to include Kotlin code in JavaScript and TypeScript projects.
+
+Top-level declarations marked with [`@JsExport`](js-to-kotlin-interop.md#jsexport-annotation) in a project that produces
+executable files (`binaries.executable()`) will get a `.d.ts` file generated, which contains the TypeScript definitions
+for the exported Kotlin declarations.
+In Kotlin 1.4, these declarations can be found in `build/js/packages/<package_name>/kotlin` alongside the corresponding,
```
Comment (koshachy) [inline docs/topics/js/js-ir-compiler.md:102]:
```suggestion
In Kotlin 1.4, these declarations can be found in `build/js/packages/<package_name>/kotlin` alongside the corresponding
```

**Review by koshachy — APPROVED** [review]
Good job!
Some minor comments.

**Comment by p7nov** [comment]
Thanks @koshachy, fixed as you suggest.

**Pull request merged**